clang/test/SemaCXX/GH93650.cpp

```diff
@@ -0,0 +1,12 @@
+// RUN: %clang_cc1 -std=c++2c -fsyntax-only -verify %s
+// expected-no-diagnostics
+
+namespace std {
+struct type_info {
+  const char *name;
+};
+} // namespace std
+
+namespace GH93650_bug {
+auto func(auto... inputArgs) { return typeid(inputArgs...[0]); }
+} // namespace GH93650_bug
```

Comment:
nit: Can you move this test to `clang/test/SemaCXX/cxx2c-pack-indexing.cpp` with a new namespace `GH93650`? I prefer avoiding scattering tests everywhere so that someone can just rerun the single test quickly when the implementation is altered in future.